File: client-app/src/models/users/UserProfile.ts
import { User } from "./User";

export interface UserProfile {
    username: string;
    displayName: string;
    imageUri?: string;
    biography?: string;
    followersCount: number;
    followingCount: number;
    isFollowing: boolean;
    images: ProfileImage[];
}

export class UserProfile implements UserProfile {
    public constructor(user: User) {
        this.username = user.username;
        this.displayName = user.displayName;
        this.imageUri = user.imageUri;
    }
}

export interface UserActivity {
    id: string;
    title: string;
    category: string;
    date: Date;
}

export interface ProfileImage {
    id: string;
    uri: string;
    isMain: boolean;
}

File: Domain/Entities/Users/UserProfile.cs
using System.ComponentModel.DataAnnotations;
using Domain.Entities.Photos;

namespace Domain.Entities.Users;

public class UserProfile
{
    [Required]
    public string Username { get; set; }

    [Required]
    public string DisplayName { get; set; }

    public string? Biography { get; set; }
    
    public string? ImageUri { get; set; }

    public bool IsFollowing { get; set; }

    public int FollowersCount { get; set; }

    public int FollowingCount { get; set; }

    public List<Image> Images { get; set; } = new();
}